### ISSUE TITLE
Media Tab "Delete" is Ambiguous

### ISSUE THREAD

**Issue by miqrogroove** [fisharebest/webtrees#3320]
I'm having some confusion with the "Delete" icons on the Media tab of the individual page.  If I use this button, it only deletes the relationship between the individual and the media object, leaving me with an orphaned media object that is difficult to find again.  This is not useful in many situations.

I think it would be helpful if this button said "Unlink" or "Remove Fact" instead of "Delete".

Even better, I would like an option or a reminder about deleting the media object completely if there are no other links to the object.

**Comment by fisharebest** [comment]
The same ambiguity exists on the “Notes” and “Sources” tabs.

**Comment by fisharebest** [comment]
For the themes that use bitmap icons (clouds, colors, webtrees, xenea), we don't have an icon for "unlink".

If we are going to introduce a new term "unlink", do we need to use "link" when we create the data initially?

e.g. "Link a media object to this individual", rather than "Add a media object"?

**Comment by miqrogroove** [comment]
Adding is constructive, whereas "Delete" is giving me orphaned objects.  So as I see it, there is more of a unique problem with media.  On the other hand, the orphaned objects are only a minor issue and would affect larger sites only.  Perhaps an alternative solution is to update the Media List page to search by objects that have no links.  Or, is "search filters" field able to do that already?

**Comment by vytux-com** [comment]
> Adding is constructive, whereas "Delete" is giving me orphaned objects. So as I see it, there is more of a unique problem with media. On the other hand, the orphaned objects are only a minor issue and would affect larger sites only. Perhaps an alternative solution is to update the Media List page to search by objects that have no links. Or, is "search filters" field able to do that already?

That would potentially introduce a privacy issue. If a media item was linked to an individual in a private tree as well as a public one the deletion of the media from the public tree and a subsequent search could infer that the individual exists in the private tree.

**Comment by miqrogroove** [comment]
I certainly wouldn't mind if there was an admin-only orphan search.

**Comment by miqrogroove** [comment]
> For the themes that use bitmap icons (clouds, colors, webtrees, xenea), we don't have an icon for "unlink".

I wouldn't even change the icon.  It's the link title that's confusing me more than anything.  I hover it and it says Delete.  I click it, and the object is not deleted.  So my personal preference would be for a title that says "Unlink" or "Remove from individual" or anything similarly descriptive.

**Comment by miqrogroove** [comment]
Still having significant problems with this.

If we can't or won't fix the button, is it possible to add a search feature to help find orphaned photos?

**Comment by AlexKalopsia** [comment]
Also struggling with this. 
I can connect new individuals one by one but - if I make a mistake - I have no idea how to remove that one individual from the list.

The generic "Delete" button is very unclear.

**Comment by fisharebest** [comment]
> The generic "Delete" button is very unclear.

@AlexKalopsia - the button was renamed from "Delete" to "Unlink".

Perhaps you are describing a different issue?
Can you give an example with screenshots?

**Comment by AlexKalopsia** [comment]
Yeah, it might be that I am doing this action in an unusual way.

I open a group photo (media object) and from there I click Edit -> Link this media object to an individual. 

Now, it happened to me that I linked the wrong individual (mostly because there are a lot of individuals with the same name, and the list doesn't suggest an ID - which would be imho really useful). 

Once this is done, I try to unlink an individual, but there is no accessible way of doing so: 

[Image](https://i.imgur.com/O6QIpEG.png)

Turns out that the solution was to go to the wrong individual's page, reach the Media tab, and then remove the image (also felt risky because I wasn't sure if it would have removed all the other links too). 

I think that a cleaner solution would be to eventually add some "Unlink" buttons like these:

[Image](https://i.imgur.com/RnODAAw.png)

**Comment by fisharebest** [comment]
TIP: you can embed images here.  Makes it easier to read than having to open links in separate tabs.  (I edited your first post to embed the image).

Links are from one record to another record.
Everywhere else, we visit the "from" end of the link to delete it.

You are asking to delete links at the "to" end.  I presume you want it for all links (e.g. to sources, notes, etc.).

A problem is that there can be several links between record.  You might have a photograph of a church linked to baptism and burial events and also to a source-citation for another event.

So, the media page will show that an individual is linked to the photo - but there are many links.  So, we would need many delete buttons.  Wouldn't this get confusing?

**Comment by AlexKalopsia** [comment]
> You are asking to delete links at the "to" end. 

Maybe this is where the ambiguity is. To me this definitely feels like a From. I even read "Link this media TO an individual".

I didn't consider that one individual could have multiple links to the same piece of media tbh. The way I currently read it (at least that's what the affordance of this page is) is "Which individuals are on this picture?", and I think this is causing the issue of me wanting to unlink from here (because this is also the place where I link to an individual).
